modules/grpc/otel/otel-protobuf-formatter.cpp

```diff
@@ -329,6 +347,57 @@ ProtobufFormatter::get_metadata(LogMessage *msg, Resource &resource, std::string
          get_scope_and_schema_url(msg, scope, scope_schema_url);
 }
 
+bool
+ProtobufFormatter::format(LogMessage *msg, LogRecord &log_record)
+{
+  gssize len;
+  const gchar *value;
+
+  value = _get_protobuf(msg, ".otel_raw.log", &len);
+  if (value)
+    return log_record.ParseFromArray(value, len);
```

Comment:
Shouldn't we allow missing fields (`ParsePartialFromArray()`)?

We don't check for completeness in the case of parsed values either.

If ParseFromArray fails, we end up producing an `LTR_ERROR`, which means we'll retry the message N times.
I think we should either allow partial messages in both parsed and unparsed cases,
or we should validate both cases and use `insert_fallback_log_record_from_log_msg()` instead of LTR_ERROR when the message is partial.

Reply:
As we talked about it IRL, I have modified the parsing and formatting to accept partial everywhere, even in source and parser.

Reply:
Thank you.